CopilotKit/packages/react-textarea/src/components/hovering-toolbar/hovering-toolbar.tsx

```diff
@@ -51,57 +53,82 @@ export const HoveringToolbar = (props: HoveringToolbarProps) => {
 
     const domRange = domSelection.getRangeAt(0);
     const rect = domRange.getBoundingClientRect();
-
+    
     // We use window = (0,0,0,0) as a signal that the selection is not in the original copilot-textarea,
     // but inside the hovering window.
     //
     // in such case, we simply do nothing.
     if (rect.top === 0 && rect.left === 0 && rect.width === 0 && rect.height === 0) {
       return;
     }
-    const minGapFromEdge = 60;
-    const verticalOffsetFromCorner = 35;
-    const horizontalOffsetFromCorner = 15;
-    let top = rect.top + window.scrollY - el.offsetHeight + verticalOffsetFromCorner;
-    // make sure top is in the viewport and not too close to the edge
-    if (top < minGapFromEdge) {
-      top = rect.bottom + window.scrollY + minGapFromEdge;
-    } else if (top + el.offsetHeight > window.innerHeight - minGapFromEdge) {
-      top = rect.top + window.scrollY - el.offsetHeight - minGapFromEdge;
+    
+    const verticalOffsetFromCorner = 0;
+    const horizontalOffsetFromCorner = 0;
+    
+    // position the toolbar below the selection
+    let top = rect.bottom + window.scrollY + verticalOffsetFromCorner;
+
+    // no space left at bottom, move up
+    if (rect.bottom + el.offsetHeight > window.innerHeight - verticalOffsetFromCorner) {
+      top = rect.top + window.scrollY - el.offsetHeight - verticalOffsetFromCorner;
     }
 
+    // position the toolbar in the center of the selection
     let left =
       rect.left + window.scrollX - el.offsetWidth / 2 + rect.width / 2 + horizontalOffsetFromCorner;
-    // make sure left is in the viewport and not too close to the edge
-    if (left < minGapFromEdge) {
-      left = minGapFromEdge;
-    } else if (left + el.offsetWidth > window.innerWidth - minGapFromEdge) {
-      left = window.innerWidth - el.offsetWidth - minGapFromEdge;
+
+    // no space left at left, move right
+    if (left < horizontalOffsetFromCorner) {
+      left = horizontalOffsetFromCorner;
+    }
+    // no space left at right, move left
+    else if (left + el.offsetWidth > window.innerWidth - horizontalOffsetFromCorner) {
+      left = window.innerWidth - el.offsetWidth - horizontalOffsetFromCorner;
     }
 
     el.style.opacity = "1";
     el.style.position = "absolute";
 
     el.style.top = `${top}px`;
     el.style.left = `${left}px`;
-  });
```

Comment:
There's no reason to recalculate it on every selection change (e.g., input in the textarea).